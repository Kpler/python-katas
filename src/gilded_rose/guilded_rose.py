class GildedRose(object):

    def __init__(self, items):
        self.items = items

    def update_quality(self):
        for item in self.items:
            if item.name != "Aged Brie" and item.name != "Backstage passes to a TAFKAL80ETC concert":
                if item.quality > 0:
                    if item.name != "Sulfuras, Hand of Ragnaros":
                        item.quality = item.quality - 1
            else:
                if item.quality < 50:
                    if item.name == "Backstage passes to a TAFKAL80ETC concert":
                        if item.sell_in < 6:
                            item.quality = item.quality + 3
                        elif item.sell_in < 11:
                            item.quality = item.quality + 2
                        else:
                            item.quality = item.quality + 1
                    else:
                        item.quality = item.quality + 1
            if item.name != "Sulfuras, Hand of Ragnaros":
                item.sell_in = item.sell_in - 1
            if item.sell_in < 0:
                if item.name != "Aged Brie":
                    if item.name != "Backstage passes to a TAFKAL80ETC concert":
                        if item.quality > 0:
                            if item.name != "Sulfuras, Hand of Ragnaros":
                                item.quality = item.quality - 1
                    else:
                        item.quality = item.quality - item.quality
                else:
                    if item.quality < 50:
                        item.quality = item.quality + 1


class Item:
    def __init__(self, name, sell_in, quality):
        self.name = name
        self.sell_in = sell_in
        self.quality = quality

    def __repr__(self):
        return "%s, %s, %s\n" % (self.name, self.sell_in, self.quality)
