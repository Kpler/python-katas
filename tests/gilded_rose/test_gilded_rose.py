from approvaltests.combination_approvals import verify_all_combinations

from src.gilded_rose.guilded_rose import Item, GildedRose


def fn_1(var_1: int, var_2: int) -> int:
    return 9091 * var_1 * var_2


def _execute_gilded_rose_update_quality() -> list[Item]:
    gilded_rose = GildedRose([])
    gilded_rose.update_quality()
    return gilded_rose.items



def test_simple():
    verify_all_combinations(
        fn_1, [list(range(20)), list(range(20))]
    )

def test_gilded_rose():
    verify_all_combinations(
        _execute_gilded_rose_update_quality,
        [],
    )
