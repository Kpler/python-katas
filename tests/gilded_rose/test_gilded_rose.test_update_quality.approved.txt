TestItem1, -1, 0
